src/vs/workbench/services/editor/browser/editorService.ts

```diff
@@ -162,10 +162,8 @@ export class EditorService extends Disposable implements EditorServiceImpl {
 					this.handleActiveEditorChange(group);
 					this._onDidVisibleEditorsChange.fire();
 					break;
-				case GroupChangeKind.EDITOR_CLOSE:
-				case GroupChangeKind.EDITOR_OPEN:
-				case GroupChangeKind.EDITOR_MOVE:
-					this._onDidEditorsChange.fire();
+				default:
+					this._onDidEditorsChange.fire({ groupId: group.id, ...e });
```

Comment:
This seems to have changed the original idea of the event as outlined by the JSDoc:

https://github.com/microsoft/vscode/blob/05cc11ed98e7dc28112c389ff61921a25e269912/src/vs/workbench/services/editor/common/editorService.ts#L99-L107

This would be a very very chatty event then given it releases any group change and editor change (including tab labels changing, dirty state, pinned state etc).

If we go down this route we need to clearly document that consumers should debounce the handling. Alternatively, wouldn't it be nicer to offer an array of `IEditorsChangeEvent` and do the micro task handling from the emitter here instead of asking all clients to do this? Similar to the `DebounceEmitter` but using micro task.

Reply:
I can filter out the chattier events we don't need. I just forwarded them all for now as I figure it out.